Remove references to `prop_index_site_code` view

esoergel opened the pull request:
@nickpell
I'm replacing these view calls with comparable SQL calls.  The tests compare the old and the new versions.  Once this is deployed, I'll remove the tests and the view itself.

esoergel committed:
Remove lookup_by_property

esoergel committed:
Get rid of site_codes_for_domain

esoergel committed:
Port Location.by_site_code to SQL

dimagimon commented on corehq/apps/locations/forms.py:
```diff
@@ -16,9 +16,9 @@
+from .models import Location, SQLLocation
 from .signals import location_created, location_edited
-from .util import load_locs_json, allowed_child_types, lookup_by_property
+from .util import load_locs_json, allowed_child_types
```
F401 'load_locs_json' imported but unused

dimagimon commented on corehq/apps/locations/tests/test_reupholster.py:
```diff
@@ -86,6 +86,54 @@ def test_filter_by_type_no_root(self):
+    ).first()
+    return Location.get(result['id']) if result else None
+
+def _sql_by_site_code(domain, site_code):
```
E302 expected 2 blank lines, found 1

esoergel committed:
pep8

nickpell commented:
looks good, waiting on tests

esoergel committed:
Exclude current location from conflict query

esoergel committed:
Merge branch 'master' of github.com:dimagi/commcare-hq into couch-location-culling

esoergel committed:
Use delete_all_locations

This was deleting SQLLocations, then attempting to lookup a couch
location, but I'd I'd changed the lookup to use SQL, so of course it
didn't work.

I also switched by_site_code to be case insensitive, although I don't
think that'll matter in practice.

esoergel commented:
@nickpell tests passed

dannyroberts commented:
good stuff!
